modules/op-improving-performance.adoc

```diff
@@ -0,0 +1,20 @@
+// This module is included in the following assemblies:
+// * resource/managing-pipelines-performance.adoc
+
+:_content-type: PROCEDURE
+[id="improving-performance_{context}"]
+= Improving {pipelines-shortname} performance
+
+If you experience slowness or recurrent failures of pipeline runs, you can take any of the following steps to improve the performance of {pipelines-shortname}.
+
+* Monitor the resource usage of the nodes in the {OCP} cluster on which {pipelines-shortname} runs. If the resource usage is high, increase the number of nodes.
++
+* Enable high-availability mode. This mode affects the controller that creates and starts pods for task runs and pipeline runs. In Red Hat testing, high-availability mode significantly reduced pipeline execution times as well as the delay from creating a `TaskRun` resource CR to the start of the pod executing the task run. To enable high-availability mode, make the following changes in the `TektonConfig` custom resource (CR):
+** Set the `pipeline.performance.disable-ha` spec to `false`.
+** Set the `pipeline.performance.buckets` spec to a number between `5` and `10`.
+** Set the `pipeline.performance.replicas` spec to a number higher than `2` and lower than or equal to the `pipeline.performance.buckets` setting.
++
+[NOTE]
```

Comment:
The first half of this statement is a note, the second half is a warning. I would suggest breaking it up accordingly. Additionally, be more specific when referring to specs. See below:

```
[NOTE]
====
You can use different variables for `pipeline.performance.buckets` and `pipeline.performance.replicas` attributes to observe the effect on performance.
====

[WARNING]
====
Exhausting the node resources or the CPU and memory availability might occur. 
====